### PR TITLE
📖[Docs] Document minimal catalog selection

### PR DESCRIPTION
**Pull request by bentito**
Documents the new features for catalog selection.

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Comment by bentito** [comment]
Addresses https://github.com/operator-framework/operator-controller/issues/1154

**Comment by dhoover103** [comment]
Honestly, I don't have anything to add - this already has various examples that I was still working on putting together for my version of the doc.

File: docs/drafts/controlling-catalog-selection.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: my-extension
+spec:
+  packageName: my-package
+  catalog:
+    selector:
+      matchExpressions:
+        - key: olm.operatorframework.io/name
```
Comment (joelanford) [inline docs/drafts/controlling-catalog-selection.md:93]:
```suggestion
        - key: olm.operatorframework.io/metadata.name
```

File: docs/drafts/controlling-catalog-selection.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: my-extension
+spec:
+  packageName: my-package
+  catalog:
+    selector:
+      matchExpressions:
+        - key: redhat.com/support
```
Comment (joelanford) [inline docs/drafts/controlling-catalog-selection.md:68]: Nit: 
```suggestion
        - key: example.com/support
```

File: docs/drafts/controlling-catalog-selection.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+         ref: quay.io/example/catalog-a:latest
+   ```
+
+   ```yaml
+   apiVersion: catalogd.operatorframework.io/v1alpha1
+   kind: ClusterCatalog
+   metadata:
+     name: catalog-b
+     labels:
+       redhat.com/support: "false"
```
Comment (bentito) [inline docs/drafts/controlling-catalog-selection.md:183]:
```suggestion
       example.com/support: "false"
```

File: docs/drafts/controlling-catalog-selection.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: my-extension
+spec:
+  packageName: my-package
+  catalog:
+    selector:
+      matchLabels:
+        redhat.com/support: "true"
```
Comment (bentito) [inline docs/drafts/controlling-catalog-selection.md:51]:
```suggestion
        example.com/support: "true"
```

File: docs/drafts/controlling-catalog-selection.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+## Important Notes
+
+- **Default Behavior**: If you do not specify any catalog selection criteria, the system may select any available catalog that provides the requested package, and the choice is undefined.
+- **Logical AND of Selectors**: When using both `matchLabels` and `matchExpressions`, catalogs must satisfy all criteria.
+- **Deprecation Status**: Non-deprecated bundles are preferred over deprecated ones during resolution.
+- **Error Messages**: The system will update the `.status.conditions` of the `ClusterExtension` with meaningful messages if resolution fails due to ambiguity or no catalogs being selected.
+
+## References
+
+- [Minimal Controls for Selecting Catalogs to Resolve From](https://github.com/operator-framework/operator-controller/issues/1028)
```
Comment (joelanford) [inline docs/drafts/controlling-catalog-selection.md:236]: Add another reference to upstream kubernetes labels and selectors doc? https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

Comment (bentito) [reply docs/drafts/controlling-catalog-selection.md:236]: done in e6864f8a

**Review by joelanford — APPROVED** [review]

**Pull request merged**